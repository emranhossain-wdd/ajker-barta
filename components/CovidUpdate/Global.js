import React, { useEffect, useState } from 'react';
import {
  LineChart,
  Line,
  XAxis,
  YAxis,
  CartesianGrid,
  Tooltip,
  Legend,
  ResponsiveContainer
} from "recharts";
import styles from '../../styles/Covid.module.css'
import { HiUsers } from 'react-icons/hi'
import { FcBusinessman, FcDataRecovery } from 'react-icons/fc'
import { BsEmojiSmile } from 'react-icons/bs'
import { RiHotelBedLine } from 'react-icons/ri'
import { AiOutlineUser } from 'react-icons/ai'

const Global = () => {
  const [globalData, setGlobalData] = useState({})
  useEffect(() => {
    fetch('https://corona.lmao.ninja/v2/all?yesterday')
      .then(res => res.json())
      .then(data => setGlobalData(data))
      .catch(err => console.log(err.message))
  }, [])
  const { cases, recovered, active, deaths, todayCases, todayRecovered } = globalData
  const covidData = [
    {
      name: "TC",
      Globaly: cases,
    },
    {
      name: "TR",
      Globaly: recovered,
    },
    {
      name: "AC",
      Globaly: active
    },
    {
      name: "TD",
      Globaly: deaths,
    },
    {
      name: "TC",
      Globaly: todayCases
    },
    {
      name: "TR",
      Globaly: todayRecovered,
    }
  ];

  return (
    <div className={styles.global}>
      <div className="container">
        <h2 className={styles.title}>Global Update</h2>
        <div className={styles.globalRow}>
          <div className={styles.lasUpdate}>
            <div className={styles.infoBox}>
              <div>
                <HiUsers style={{ color: "#18dcff", fontSize: "25px", margin: "0 auto" }} />
                <p>Total Cases</p>
                <h1>{cases}</h1>
              </div>
            </div>
            <div className={styles.infoBox}>
              <div>
                <BsEmojiSmile style={{ color: "#feca57", fontSize: "25px", margin: "0 auto" }} />
                <p>Total Recovered</p>
                <h1>{recovered}</h1>
              </div>
            </div>
            <div className={styles.infoBox}>
              <div>
                <FcBusinessman style={{ fontSize: "25px", margin: "0 auto" }} />
                <p>Active Cases</p>
                <h1>{active}</h1>
              </div>
            </div>
            <div className={styles.infoBox}>
              <div>
                <RiHotelBedLine style={{ color: "#0be881", fontSize: "25px", margin: "0 auto" }} />
                <p>Total Deaths</p>
                <h1>{deaths}</h1>
              </div>
            </div>
            <div className={styles.infoBox}>
              <div>
                <AiOutlineUser style={{ color: "#f368e0", fontSize: "25px", margin: "0 auto" }} />
                <p>Todays Case</p>
                <h1>{todayCases}</h1>
              </div>
            </div>
            <div className={styles.infoBox}>
              <div>
                <FcDataRecovery style={{ fontSize: "25px", margin: "0 auto" }} />
                <p>Todays Recovered</p>
                <h1>{todayRecovered}</h1>
              </div>
            </div>
          </div>
          <div>
            <div className={styles.chart}>
              <ResponsiveContainer>
                <LineChart className={styles.lineChart} data={covidData}>
                  <CartesianGrid strokeDasharray="3 3" />
                  <XAxis dataKey="name" padding={{ left: 30, right: 30 }} />
                  <YAxis style={{ fontSize: "11px" }} />
                  <Tooltip />
                  <Legend />
                  <Line
                    type="monotone"
                    dataKey="Globaly"
                    stroke="#8884d8"
                    activeDot={{ r: 8 }}
                  />
                </LineChart>
              </ResponsiveContainer>
            </div>
            <div className={styles.indicateColor}>
              <div><div style={{ color: "#18dcff", fontWeight: "500" }}>TC</div>Total Cases</div>
              <div><div style={{ color: "#feca57", fontWeight: "500" }}>TR</div>Total Recovered</div>
              <div><div style={{ color: "#ff4d4d", fontWeight: "500" }}>AC</div>Active Casess</div>
              <div><div style={{ color: "#0be881", fontWeight: "500" }}>TD</div>Total Deaths</div>
              <div><div style={{ color: "#f368e0", fontWeight: "500" }}>TC</div>Todays Case</div>
              <div><div style={{ color: "#3c40c6", fontWeight: "500" }}>TR</div>Todays Recovered</div>
            </div>
          </div>
        </div>
      </div>
    </div>
  );
};

export default Global;