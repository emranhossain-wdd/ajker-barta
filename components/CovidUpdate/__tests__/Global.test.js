import React from 'react'
import ReactDOM from 'react-dom'
import renderer from 'react-test-renderer'
import Global from '../Global'

it('render Global component without crashing', () => {
  const div = document.createElement('div')
  ReactDOM.render(<Global />, div)
})
it('maching snapshot', () => {
  const tree = renderer.create(<Global />)
  expect(tree).toMatchSnapshot()
})