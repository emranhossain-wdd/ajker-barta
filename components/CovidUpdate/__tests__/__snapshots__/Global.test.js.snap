// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`maching snapshot 1`] = `
<div
  className="global"
>
  <div
    className="container"
  >
    <h2
      className="title"
    >
      Global Update
    </h2>
    <div
      className="globalRow"
    >
      <div
        className="lasUpdate"
      >
        <div
          className="infoBox"
        >
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "#18dcff",
                  "fontSize": "25px",
                  "margin": "0 auto",
                }
              }
              viewBox="0 0 20 20"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M9 6a3 3 0 11-6 0 3 3 0 016 0zM17 6a3 3 0 11-6 0 3 3 0 016 0zM12.93 17c.046-.327.07-.66.07-1a6.97 6.97 0 00-1.5-4.33A5 5 0 0119 16v1h-6.07zM6 11a5 5 0 015 5v1H1v-1a5 5 0 015-5z"
              />
            </svg>
            <p>
              Total Cases
            </p>
            <h1 />
          </div>
        </div>
        <div
          className="infoBox"
        >
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "#feca57",
                  "fontSize": "25px",
                  "margin": "0 auto",
                }
              }
              viewBox="0 0 16 16"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M8 15A7 7 0 1 1 8 1a7 7 0 0 1 0 14zm0 1A8 8 0 1 0 8 0a8 8 0 0 0 0 16z"
              />
              <path
                d="M4.285 9.567a.5.5 0 0 1 .683.183A3.498 3.498 0 0 0 8 11.5a3.498 3.498 0 0 0 3.032-1.75.5.5 0 1 1 .866.5A4.498 4.498 0 0 1 8 12.5a4.498 4.498 0 0 1-3.898-2.25.5.5 0 0 1 .183-.683zM7 6.5C7 7.328 6.552 8 6 8s-1-.672-1-1.5S5.448 5 6 5s1 .672 1 1.5zm4 0c0 .828-.448 1.5-1 1.5s-1-.672-1-1.5S9.448 5 10 5s1 .672 1 1.5z"
              />
            </svg>
            <p>
              Total Recovered
            </p>
            <h1 />
          </div>
        </div>
        <div
          className="infoBox"
        >
          <div>
            <svg
              enableBackground="new 0 0 48 48"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "fontSize": "25px",
                  "margin": "0 auto",
                }
              }
              version="1"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <polygon
                fill="#FF9800"
                points="24,37 19,31 19,25 29,25 29,31"
              />
              <g
                fill="#FFA726"
              >
                <circle
                  cx="33"
                  cy="19"
                  r="2"
                />
                <circle
                  cx="15"
                  cy="19"
                  r="2"
                />
              </g>
              <path
                d="M33,13c0-7.6-18-5-18,0c0,1.1,0,5.9,0,7c0,5,4,9,9,9s9-4,9-9C33,18.9,33,14.1,33,13z"
                fill="#FFB74D"
              />
              <path
                d="M24,4c-6.1,0-10,4.9-10,11c0,0.8,0,2.3,0,2.3l2,1.7v-5l12-4l4,4v5l2-1.7c0,0,0-1.5,0-2.3c0-4-1-8-6-9l-1-2 H24z"
                fill="#424242"
              />
              <g
                fill="#784719"
              >
                <circle
                  cx="28"
                  cy="19"
                  r="1"
                />
                <circle
                  cx="20"
                  cy="19"
                  r="1"
                />
              </g>
              <polygon
                fill="#fff"
                points="24,43 19,31 24,32 29,31"
              />
              <polygon
                fill="#D32F2F"
                points="23,35 22.3,39.5 24,43.5 25.7,39.5 25,35 26,34 24,32 22,34"
              />
              <path
                d="M29,31L29,31l-5,12l-5-12c0,0-11,2-11,13h32C40,33,29,31,29,31z"
                fill="#546E7A"
              />
            </svg>
            <p>
              Active Cases
            </p>
            <h1 />
          </div>
        </div>
        <div
          className="infoBox"
        >
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "#0be881",
                  "fontSize": "25px",
                  "margin": "0 auto",
                }
              }
              viewBox="0 0 24 24"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g>
                <path
                  d="M0 0h24v24H0z"
                  fill="none"
                />
                <path
                  d="M22 11v9h-2v-3H4v3H2V4h2v10h8V7h6a4 4 0 0 1 4 4zm-2 3v-3a2 2 0 0 0-2-2h-4v5h6zM8 11a1 1 0 1 0 0-2 1 1 0 0 0 0 2zm0 2a3 3 0 1 1 0-6 3 3 0 0 1 0 6z"
                />
              </g>
            </svg>
            <p>
              Total Deaths
            </p>
            <h1 />
          </div>
        </div>
        <div
          className="infoBox"
        >
          <div>
            <svg
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": "#f368e0",
                  "fontSize": "25px",
                  "margin": "0 auto",
                }
              }
              viewBox="0 0 1024 1024"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                d="M858.5 763.6a374 374 0 0 0-80.6-119.5 375.63 375.63 0 0 0-119.5-80.6c-.4-.2-.8-.3-1.2-.5C719.5 518 760 444.7 760 362c0-137-111-248-248-248S264 225 264 362c0 82.7 40.5 156 102.8 201.1-.4.2-.8.3-1.2.5-44.8 18.9-85 46-119.5 80.6a375.63 375.63 0 0 0-80.6 119.5A371.7 371.7 0 0 0 136 901.8a8 8 0 0 0 8 8.2h60c4.4 0 7.9-3.5 8-7.8 2-77.2 33-149.5 87.8-204.3 56.7-56.7 132-87.9 212.2-87.9s155.5 31.2 212.2 87.9C779 752.7 810 825 812 902.2c.1 4.4 3.6 7.8 8 7.8h60a8 8 0 0 0 8-8.2c-1-47.8-10.9-94.3-29.5-138.2zM512 534c-45.9 0-89.1-17.9-121.6-50.4S340 407.9 340 362c0-45.9 17.9-89.1 50.4-121.6S466.1 190 512 190s89.1 17.9 121.6 50.4S684 316.1 684 362c0 45.9-17.9 89.1-50.4 121.6S557.9 534 512 534z"
              />
            </svg>
            <p>
              Todays Case
            </p>
            <h1 />
          </div>
        </div>
        <div
          className="infoBox"
        >
          <div>
            <svg
              enableBackground="new 0 0 48 48"
              fill="currentColor"
              height="1em"
              stroke="currentColor"
              strokeWidth="0"
              style={
                Object {
                  "color": undefined,
                  "fontSize": "25px",
                  "margin": "0 auto",
                }
              }
              version="1"
              viewBox="0 0 48 48"
              width="1em"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                fill="#D1C4E9"
              >
                <path
                  d="M38,7H10C8.9,7,8,7.9,8,9v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2V9C40,7.9,39.1,7,38,7z"
                />
                <path
                  d="M38,19H10c-1.1,0-2,0.9-2,2v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2v-6C40,19.9,39.1,19,38,19z"
                />
                <path
                  d="M38,31H10c-1.1,0-2,0.9-2,2v6c0,1.1,0.9,2,2,2h28c1.1,0,2-0.9,2-2v-6C40,31.9,39.1,31,38,31z"
                />
              </g>
              <g
                fill="#F44336"
              >
                <rect
                  height="20"
                  width="6"
                  x="35"
                  y="28"
                />
                <rect
                  height="6"
                  width="20"
                  x="28"
                  y="35"
                />
              </g>
            </svg>
            <p>
              Todays Recovered
            </p>
            <h1 />
          </div>
        </div>
      </div>
      <div>
        <div
          className="chart"
        >
          <div
            className="recharts-responsive-container"
            style={
              Object {
                "height": "100%",
                "maxHeight": undefined,
                "minHeight": undefined,
                "minWidth": undefined,
                "width": "100%",
              }
            }
          />
        </div>
        <div
          className="indicateColor"
        >
          <div>
            <div
              style={
                Object {
                  "color": "#18dcff",
                  "fontWeight": "500",
                }
              }
            >
              TC
            </div>
            Total Cases
          </div>
          <div>
            <div
              style={
                Object {
                  "color": "#feca57",
                  "fontWeight": "500",
                }
              }
            >
              TR
            </div>
            Total Recovered
          </div>
          <div>
            <div
              style={
                Object {
                  "color": "#ff4d4d",
                  "fontWeight": "500",
                }
              }
            >
              AC
            </div>
            Active Casess
          </div>
          <div>
            <div
              style={
                Object {
                  "color": "#0be881",
                  "fontWeight": "500",
                }
              }
            >
              TD
            </div>
            Total Deaths
          </div>
          <div>
            <div
              style={
                Object {
                  "color": "#f368e0",
                  "fontWeight": "500",
                }
              }
            >
              TC
            </div>
            Todays Case
          </div>
          <div>
            <div
              style={
                Object {
                  "color": "#3c40c6",
                  "fontWeight": "500",
                }
              }
            >
              TR
            </div>
            Todays Recovered
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
